cmd/cspi-mgmt/app/handler.go

```diff
@@ -241,7 +253,7 @@ func (c *CStorPoolInstanceController) removeFinalizer(cspi *apis.CStorPoolInstan
 	if len(cspi.Finalizers) == 0 {
 		return nil
 	}
-	cspi.Finalizers = []string{}
+	cspi.Finalizers = util.RemoveString(cspi.Finalizers, apiscspc.PoolProtectionFinalizer)
```

Comment:
hope this works even if cspi.Finalizers doesn't have PoolProtectionFinalizer

Reply:
It works in both cases when the cspi.Finalizers slice is nil or the PoolProtectionFinalizer is not present in the slice. https://play.golang.org/p/kgnN2Xv_U4r